Textile/Classes/Public/TextileApi.m

```diff
@@ -206,11 +209,15 @@ - (void)createNodeDependants {
   self.profile = [[ProfileApi alloc] initWithNode:self.node];
   self.schemas = [[SchemasApi alloc] initWithNode:self.node];
   self.threads = [[ThreadsApi alloc] initWithNode:self.node];
+}
 
-  self.lifecycleManager = [[LifecycleManager alloc] initWithNode:self.node];
-  self.lifecycleManager.delegate = self.delegate;
+- (void)forLater {
+  // depending on the context, we can change the the implementation that gets assigned here.
+  self.lifecycleManager = [[TTEAppStateLifecycleManager alloc] initWithTextile:self];
```

Comment:
This is were we could use dependency injection to use a different implementation of `TTELifecyceManager`